searx/searx
# Add Docker Hub search engine.

![MySQL Docker image](https://user-images.githubusercontent.com/25624882/107155963-8b910c00-6949-11eb-98c0-2e01e535cdcb.png)
![Searx Docker image](https://user-images.githubusercontent.com/25624882/107156131-79639d80-694a-11eb-9a27-6af3e2b458e5.png)
## What does this PR do?
This adds [Docker Hub](https://hub.docker.com) to Searx as a search engine and resolves issue #2433. I modeled this engine around the existing SoundCloud engine, the engine's favicon was downloaded from the Docker Hub website with `wget` and converted to a PNG with ImageMagick, and the shortcut was plucked from DDG's bangs database. It supports the parsing of URLs, titles, content, published dates, and thumbnails of Docker images.

## Why is this change important?
Once again, I wanted to resolve issue #2433. As a semi-regular user of the Hub myself, I'm sure many developers would find the direct ability to search for Docker images very convenient. 

## How to test this PR locally?
This PR is fully compatible with `make test`. To see it in action:
* `make run`
* query `!dockerhub searx`

## Related issues
Closes #2433
